### PR TITLE
Identify scenarios through testgrid and Run multiple scenario Repos

### PR DESCRIPTION
**Pull request by tharindu1992**
**Purpose**

This is to run tests from multiple test repos and Identify the executed scenarios after the completion of test run

**Security checks**
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: common/src/main/java/org/wso2/testgrid/common/config/ScenarioConfig.java [hunk 1]
```diff
@@ -107,5 +160,61 @@ public String getRemoteBranch() {
...
+    public void setFile(String file) {
+        this.file = file;
+    }
+
+    public String getOutputDir() {
+        if (StringUtil.isStringNullOrEmpty(outputDir)) {
+            try {
+                URL url = new URL(remoteRepository);
+                String [] splitbyDash = url.getPath().split("/");
+                String [] splitByDot = splitbyDash[splitbyDash.length - 1].split("\\.");
```
Comment (kasunbg) [inline common/src/main/java/org/wso2/testgrid/common/config/ScenarioConfig.java]: This is an unusual logic, and hence not very portable probably. What assumptions did you make here?
For instance, did you assume the remoteRepository will always be a 'https' url?

File: core/src/main/java/org/wso2/testgrid/core/ScenarioExecutor.java [hunk 2]
```diff
@@ -121,18 +121,11 @@ public void execute(TestScenario testScenario, DeploymentCreationResult deployme
             } else {
                 for (Test test : tests) {
                     logger.info(StringUtil.concatStrings("Executing ", test.getTestName(), " Test"));
-                    test.execute(testLocation, deploymentCreationResult);
+                    //test.execute(testLocation, deploymentCreationResult);
```
Comment (kasunbg) [inline core/src/main/java/org/wso2/testgrid/core/ScenarioExecutor.java]: commented out code.

File: core/src/main/java/org/wso2/testgrid/core/TestPlanExecutor.java [hunk 3]
```diff
@@ -282,213 +280,123 @@ private void performPostTestPlanActions(TestPlan testPlan, DeploymentCreationRes
...
+                    }
+                    if (file == null) {
+                        file = Paths.get("test.sh");
                     }
-                    break;
+                    testExecutor.init(parent.toString(), scenarioConfig.getName(), scenarioConfig);
+                    testExecutor.execute(file.toString(), deploymentCreationResult);
                 }
+
+
```
Comment (kasunbg) [inline core/src/main/java/org/wso2/testgrid/core/TestPlanExecutor.java]:
```suggestion
```

File: core/src/main/java/org/wso2/testgrid/core/TestPlanExecutor.java [hunk 4]
```diff
@@ -282,213 +280,123 @@ private void performPostTestPlanActions(TestPlan testPlan, DeploymentCreationRes
...
-                    "Error occurred while persisting test scenario ", testScenario.getName()), e);
+
+        logger.info("--------- Identified Scenarios ---------------------");
+        for (TestScenario scenario : scenarioConfig.getScenarios()) {
+            logger.info(scenario.getName());
         }
+        logger.info("-------------------------------------------------");
+    }
+
+    private void appendScenario (TestPlan testPlan, String scenarioName, ScenarioConfig scenarioConfig) {
```
Comment (kasunbg) [inline core/src/main/java/org/wso2/testgrid/core/TestPlanExecutor.java]:
```suggestion
    private void appendScenario(TestPlan testPlan, String scenarioName, ScenarioConfig scenarioConfig) {
```

File: core/src/test/java/org/wso2/testgrid/core/command/RunTestPlanCommandTest.java [hunk 5]
```diff
@@ -177,6 +177,10 @@ private void copyWorkspaceArtifacts() throws IOException {
         String destinationDir = Paths.get(workspaceDir, "/workspace").toString();
         FileUtils.copyDirectory(new File(resourceDir), new File(destinationDir));
         logger.info("Copying necessary artifacts to directory: " + destinationDir);
+        resourceDir = Paths.get("./src", "test", "resources", "workspace", "data-bucket").toString();
```
Comment (kasunbg) [inline core/src/test/java/org/wso2/testgrid/core/command/RunTestPlanCommandTest.java]:
```suggestion
        resourceDir = Paths.get("src", "test", "resources", "workspace", "data-bucket").toString();
```

File: infrastructure/src/main/java/org/wso2/testgrid/infrastructure/providers/ShellScriptProvider.java [hunk 6]
```diff
@@ -60,27 +61,30 @@ public void init(TestPlan testPlan) throws TestGridInfrastructureException  {
...
+                            String testInputsLoc = DataBucketsHelper.getInputLocation(testPlan)
+                                    .toAbsolutePath().toString();
+                            final String command = "bash " + script.getFile() + " --input-dir " + testInputsLoc;
+                            int exitCode = shellExecutor.executeCommand(command);
+                            if (exitCode > 0) {
+                                throw new TestGridInfrastructureException(StringUtil.concatStrings(
+                                        "Error while executing ", script.getFile(),
+                                        ". Script exited with a non-zero exit code (exit code = ", exitCode, ")"));
+                            }
+                        } catch (CommandExecutionException e) {
```
Comment (kasunbg) [inline infrastructure/src/main/java/org/wso2/testgrid/infrastructure/providers/ShellScriptProvider.java]: Shouldn't we run rest of the scenario configs if there was some other exception..

File: web/src/main/react-dashboard/package-lock.json [hunk 7]
```diff
@@ -14,7 +14,7 @@
       "resolved": "https://registry.npmjs.org/accepts/-/accepts-1.3.4.tgz",
```
Comment (kasunbg) [inline web/src/main/react-dashboard/package-lock.json]: There are major changes to this file. Can you verify and revert any unnecessary changes? I don't see any dashboard related changes in this pr.

File: common/src/main/java/org/wso2/testgrid/common/TestGridConstants.java [hunk 8]
```diff
@@ -69,7 +69,7 @@
      */
     public static final String TEST_TYPE_FUNCTIONAL = "FUNCTIONAL";
     public static final String TEST_TYPE_PERFORMANCE = "PERFORMANCE";
-    public static final String TEST_TYPE_INTEGRATION = "INTEGRATION";
+    public static final String TEST_TYPE_INTEGRATION = "TESTNG";
```
[the remark this answers is not in the document]
Comment (tharindu1992) [reply common/src/main/java/org/wso2/testgrid/common/TestGridConstants.java]: Yes we will have to update all testgrid.yaml files

File: automation/src/main/java/org/wso2/testgrid/automation/parser/TestNgResultsParser.java [hunk 9]
```diff
@@ -120,10 +121,16 @@ public TestNgResultsParser(TestScenario testScenario, String testLocation) {
      */
     @Override
     public void parseResults() throws ResultParserException {
-        final Path dataBucket = DataBucketsHelper.getOutputLocation(testScenario.getTestPlan());
+        Path dataBucket = DataBucketsHelper.getOutputLocation(testScenario.getTestPlan());
+        dataBucket = Paths.get(dataBucket.toString(), "test-outputs/scenarios", testScenario.getOutputDir(),
```
Comment (azinneera) [inline automation/src/main/java/org/wso2/testgrid/automation/parser/TestNgResultsParser.java]: "test-outputs/scenarios" can be replaced with TestGridConstants.TTEST_RESULTS_DIR

File: automation/src/main/java/org/wso2/testgrid/automation/parser/TestNgResultsParser.java [hunk 10]
```diff
@@ -120,10 +121,16 @@ public TestNgResultsParser(TestScenario testScenario, String testLocation) {
...
+                testScenario.getName());
+        logger.info(testScenario.getName());
         Set<Path> inputFiles = getResultInputFiles(dataBucket);
 
-        final Path outputLocation = DataBucketsHelper.getOutputLocation(testScenario.getTestPlan());
+        Path outputLocation = DataBucketsHelper.getOutputLocation(testScenario.getTestPlan());
+        outputLocation = Paths.get(outputLocation.toString(), "test-outputs/scenarios",
```
Comment (azinneera) [inline automation/src/main/java/org/wso2/testgrid/automation/parser/TestNgResultsParser.java]: `test-outputs/scenarios` can be replaced with `TestGridConstants.TEST_RESULTS_DIR`

File: automation/src/test/java/org/wso2/testgrid/automation/executor/TestNgResultsParserTest.java [hunk 11]
```diff
@@ -44,27 +44,35 @@
...
 import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Optional;
 
 public class TestNgResultsParserTest {
 
     private static final String TESTGRID_HOME = Paths.get("target", "testgrid-home").toString();
     private static final String SUREFIRE_REPORTS_DIR = "surefire-reports";
+    private static final String SUREFIRE_REPORTS_DIR_OUT = "test-outputs/scenarios/SolutionPattern22/surefire-reports";
```
Comment (azinneera) [inline automation/src/test/java/org/wso2/testgrid/automation/executor/TestNgResultsParserTest.java]: `test-outputs/scenarios` can be replaced with `TestGridConstants.TEST_RESULTS_DIR`

File: common/src/main/java/org/wso2/testgrid/common/config/ScenarioConfig.java [hunk 12]
```diff
@@ -81,6 +129,11 @@ public void setTestType(String testType) {
         this.testType = testType;
     }
 
+    private Map<String, String> inputParameters = new HashMap<>();
+
+
```
Comment (azinneera) [inline common/src/main/java/org/wso2/testgrid/common/config/ScenarioConfig.java]: Unnecessary newline
